### DISCUSSION TITLE
RuntimeError: python3.9: pixels are not authentic `...' @ error/cache.c/QueueAuthenticPixelCacheNexus/4187

### DISCUSSION THREAD

**Discussion by jackthenewbie** [ImageMagick/ImageMagick#6243] [General]
### ImageMagick version

7

### Operating system, version and so on

Debian 11

### Description
Error: ``RuntimeError: python3.9: pixels are not authentic `Image/Assets/ground_back/Electro_back.png' @ error/cache.c/QueueAuthenticPixelCacheNexus/4187``
I know this may sound weird, but I'm trying to hook my cpp to python, so python can use function from cpp. In cpp I has:
```cpp
image->read("Image/Assets/ground_back/"+element+"_back.png");
//image->resize(Geometry(1280, 720));
Image gacha_splash(l.character(it->second.get_name())); //l.character(it->second.get_name()) generate valid url
image->draw(DrawableCompositeImage(coordinate["drawBackground"]["xsplash"].asInt(), 
                                   coordinate["drawBackground"]["ysplash"].asInt(), 0, 0, gacha_splash, OverCompositeOp));
Image ImageGround_front("Image/Assets/ground_front/"+element+"_front.png");
image->draw(DrawableCompositeImage(coordinate["drawBackground"]["shiftFront"].asInt(), 0, 0, 0, ImageGround_front, OverCompositeOp));
```
function such as ``coordinate["drawBackground"]["shiftFront"].asInt()`` it just returning an integer. The code run fine with executable compiled cpp project, but when I tried import the function contain the above code to python, it throw me that error. What could be the reasons behind this?

Edit: I try anothrt imagr instead of ``element+"_back.png"``, it run, but ye could really because of ``imagename_back.png`` corrupted (I said could, because i can run excutable file with no problem)

**Comment by snibgo** [comment]
Which of those statements caused the error?

You don't seem to have any error checking.

Did your `image->read` succeed? When you run it in Python, is the current directory the same as when you run it at the command line?

**Comment by jackthenewbie** [comment]
> Did your `image->read` succeed? When you run it in Python, is the current directory the same as when you run it at the command line?

``image->read`` is where the error happen, apparently my image somehow corrupted (even though I can access it image viewer with no problem). I even try absolute path for the image, but still got that error. But I got working set of images now, thanks:)
They were images I export from pts, I'll see if I can reproduce images, will update on this later.
